src/connection.c

```diff
@@ -212,7 +238,11 @@ enum OperationReturnCode connection_configure(struct ldap_global_context_t *glob
         get_ldap_option(connection->ldap, LDAP_OPT_X_SASL_AUTHZID, &connection->ldap_defaults->authzid);
 
         connection->ldap_defaults->flags = config->sasl_options->sasl_flags;
-        connection->ldap_defaults->mechanism = talloc_strdup(global_ctx->talloc_ctx, config->sasl_options->mechanism);
+
+        if (config->sasl_options->mechanism)
+        {
+            ld_talloc_strdup_e(connection->ldap_defaults->mechanism, error_exit, "Error - out of memory - unable to allocate memory for sasl mechanism", talloc_ctx, config->sasl_options->mechanism);
```

Comment:
In `ldap.h` there are following constants declared:
```
/* sasl methods */
#define LDAP_SASL_SIMPLE	((char*)0)
#define LDAP_SASL_NULL		("")
```

And therefore we can have `NULL` sasl mechanism using `strdup` here however may be not the best design choice. We must think of a better solution.

Reply:
Or add special behavior of assigning `NULL` in case of `NULL` value.